include/sentry.h

```diff
@@ -1284,6 +1284,20 @@ SENTRY_EXPERIMENTAL_API void sentry_transaction_set_sampled(
 SENTRY_EXPERIMENTAL_API void sentry_transaction_remove_sampled(
     sentry_value_t transaction);
 
+/**
+ * Starts a new transaction based on the provided context, restored
+ * from an external integration (i.e. a span from a different SDK)
+ * or manually constructed by a user.
+ */
+SENTRY_EXPERIMENTAL_API sentry_value_t sentry_transaction_start(
+    sentry_value_t transaction);
```

Comment:
after todays discussion are we going roll this function into `sentry_value_new_transaction`?

Reply:
i don't think we can/should roll this into `sentry_value_new_transaction`, unfortunately. i do think we may need to re-append `_context` to some of the methods if we're going ahead with this rename: a **transaction context** is at best a subset of what a full transaction contains, and some functions(/methods) on transactions would not work on a transaction context, and vice versa.

some examples:
- transaction sampling: whether a transaction will be sent or not (i.e. its `sampled` field is true/false) needs to be decided upon during transaction start
  - this is an immutable internal field on a transaction
  - this is a mutable field on a transaction context 
- `tag`s and `data`
  - transactions have these
  - transaction contexts do not

Reply:
> tags and data
> * transactions have these
> * transaction contexts do not

How would you end up adding these to a transaction then? Would they only end up on the transaction based on whatever is set on the scope?

Reply:
there should be functions to set data and tags on the Transaction itself, i.e. what's returned by `sentry_transaction_start()`. that's implemented in https://github.com/getsentry/sentry-native/pull/626.

`transaction_context`s on the other hand shouldn't have any of that information, nor should they offer the ability to set those. they're only really there so a user can set some values that are immutable after a Transaction has been started. to be fair, the `name` and `operation` on a `transaction_context` are exceptions to that rule, but it otherwise applies to everything else on a `transaction_context`.